app/controllers/course_user_data_controller.rb

```diff
@@ -136,13 +139,21 @@ def update
     end
   end
 
-  action_auth_level :destroy, :instructor
+  action_auth_no_course :destroy
   def destroy
-    @destroyCUD = @course.course_user_data.find(params[:id])
-    if @destroyCUD && @destroyCUD != @cud && params[:yes1] && params[:yes2] && params[:yes3]
-      @destroyCUD.destroy # awwww!!!
+    @course = Course.find_by(name: params[:course_name])
+    @cud = CourseUserDatum.unscoped.find(params[:id])
+
+    if @cud.instructor?
+      @destroyCUD = @course.course_user_data.find(params[:id])
```

Comment:
I think this code is incorrect.  It's using params[:id] to load 2 different cuds, but really it loads the same cud twice.  We shouldn't ever load @cud, because it gets defined by session storage and represents the actual user, not any models that we're messing with.